.gitignore

```diff
@@ -2,33 +2,353 @@
 /build
 AssemblyInfo.cs
 
-# Standard VS.NET and ReSharper Foo
-.vs/*
-src/*/obj
-src/*/bin
-samples/*/obj
-samples/*/bin
-*.csproj.user
-*ReSharper.user
-_ReSharper*
-*.suo
 *.cache
 * Thumbs.db
 
 # Leftovers from merging
 *.orig
 *.bak
-*.sln.DotSettings.user
-*.DotSettings.user
-
-# Buildscripts artifacts
-buildscripts/**/obj/
-buildscripts/**/bin/
 
 # Tools artifacts
-tools/**/obj/
```

Comment:
All the removed lines were redundant after adding the standard gitignores below.

Reply:
I think it would be better to keep the `.gitignore` overhaul out of this PR so it stays focused on the actual problem you want to fix.

(Also, in my opinion, after adding all the stock entries below there will be *more* superfluous lines, not fewer. For example, we know that Windsor isn't a ClickOnce project, nor has any C++ artifacts, so the lines for those aren't needed.)

Reply:
Happy to do so. 

The list is a standard and handling it this way makes for easy updates over time as VS and common tools change. I don't think I've ever seen a project remove parts of it. The virtue of cleaning up the manual section is that everything should just work and we're can focus on what is unique to this project. 

Would you prefer I drop the commit?

Reply:
Not necessarily drop (I personally don't mind a standard `.gitignore` for the reason you gave, but this is for @jonorossi to decide), but if it is not too much work, submit it as a separate PR of its own.

Reply:
Yes, very happy to do that. https://github.com/castleproject/Windsor/pull/441